src/F2x/grammar/fortran.g

```diff
@@ -655,7 +655,7 @@ signed_real_literal_constant
     ;
 
 real_constant_number
-    : T_DIGIT_STRING T_PERIOD_EXPONENT
+    : T_DIGIT_STRING T_PERIOD_EXPONENT (T_UNDERSCORE kind_param)?
```

Comment:
This will break the grammar as the underscore will introduce ambiguities. (There is a reason, the kind parameters are commented out in the `real_literal_constant` rule.